prisma/schema.prisma

```diff
@@ -105,3 +108,54 @@ model Attachment {
   @@index([filePath], name: "IX_Attachments_filePath")
   @@map("Attachments")
 }
+
+model ActivityLog {
+  id                   String                @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
+  createdAt            DateTime              @default(now())
+  taskId               String                @db.Uuid
+  workspaceId          String                @db.VarChar(32)
+  activityType         ActivityType
+  createTaskTracker    CreateTaskTracker?
+  assigneeTracker      AssignTracker?
+  workflowStateTracker WorkflowStateTracker?
+  task                 Task                  @relation(fields: [taskId], references: [id])
+
+  @@index([createdAt])
+  @@map("ActivityLogs")
+}
+
+model CreateTaskTracker {
+  id            String      @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
+  activityLogId String      @unique @db.Uuid
+  createdBy     String
+  createdById   String      @db.Uuid
+  activityLog   ActivityLog @relation(fields: [activityLogId], references: [id])
+}
+
+model AssignTracker {
+  id            String      @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
+  activityLogId String      @unique @db.Uuid
+  initiator     String
+  initiatorId   String      @db.Uuid
+  assignedTo    String
+  assignedToId  String      @db.Uuid
+  activityLog   ActivityLog @relation(fields: [activityLogId], references: [id])
+}
+
+model WorkflowStateTracker {
+  id                     String         @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
+  activityLog            ActivityLog    @relation(fields: [activityLogId], references: [id])
+  activityLogId          String         @unique @db.Uuid
+  initiator              String
+  initiatorId            String         @db.Uuid
+  prevWorkflowStateId    String?        @db.Uuid
+  currentWorkflowStateId String         @db.Uuid
+  prevWorkflowState      WorkflowState? @relation(name: "RL_WorkflowStateTracker_prev_workflowState", fields: [prevWorkflowStateId], references: [id])
+  currentWorkflowState   WorkflowState  @relation(name: "RL_WorkflowStateTracker_current_workflowState", fields: [currentWorkflowStateId], references: [id])
+}
+
```

Comment:
Are these new tracker tables really necessary? Could you please explain the flow behind them @aatbip